### PR TITLE
Add DelayNs impl

### PR DESCRIPTION
**Pull request by kaidokert**
## Summary by Sourcery

Implement an optional embedded-hal 1.0 DelayNs-based blocking delay API on top of the existing SysTick-based Timer and refactor internal cycle snapshot logic to support it.

New Features:
- Add an optional embedded-hal 1.0-based DelayNs implementation for Timer and &Timer using the SysTick-backed timer.
- Provide a new delay_hal example demonstrating DelayNs usage on Cortex-M via semihosting.

Enhancements:
- Refactor timer cycle capture into a reusable stable_cycle_snapshot helper and track the SysTick input frequency on Timer when the embedded-hal feature is enabled.
- Validate that the SysTick frequency is non-zero when constructing Timer instances to prevent misconfiguration.

Build:
- Introduce an optional embedded-hal feature, wire it into Cargo features, docs.rs configuration, and the qemu run alias, and register a new delay_hal example behind the appropriate feature flags.

Documentation:
- Update README and crate-level documentation to describe the new embedded-hal DelayNs support and how to enable and run the corresponding example.

Tests:
- Add unit tests for delay-cycle conversion rounding and saturation behavior, and for DelayNs zero-delay acceptance and full-duration waiting from partial tick phases.

## Summary by CodeRabbit

* **New Features**
  * Optional embedded-hal 1.0 DelayNs support when the embedded-hal feature is enabled.
  * New embedded delay example demonstrating Timer usage via the DelayNs interface.

* **Documentation**
  * README clarified (Cortex-M heading, new embedded-hal demo/instructions, adjusted TODO wording).
  * Documentation build metadata updated to include the embedded-hal feature.

* **Tests**
  * Added unit tests covering delay behavior under the embedded-hal feature.

* **Chores**
  * Tooling run alias updated to include the embedded-hal feature.

### CODE REVIEW

**Comment by kaidokert** [comment]
@sourcery-ai review

File: systick-timer/src/timer.rs [hunk 1]
```diff
@@ -157,6 +169,36 @@ impl Timer {
...
+    #[cfg(feature = "embedded-hal")]
+    fn delay_cycles_blocking(&self, cycles: u64) {
+        if cycles == 0 {
+            return;
+        }
+
+        let start = self.current_cycles_wrapping();
+        while self.current_cycles_wrapping().wrapping_sub(start) < cycles {
+            core::hint::spin_loop();
+        }
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply systick-timer/src/timer.rs:204]: this is handled in comments
